### DISCUSSION TITLE
Update node from background?

### DISCUSSION THREAD

**Discussion by bnssoftware** [xyflow/xyflow#3093] [Q&A] [answered]
Is there a way for a diagram to long pool a backend service to dynamically update the node states, such as colors, text etc?  The idea is I want to have a flowchart that gives real time feedback as to where the process is at by turning that node green.

**Comment by chrtze** [comment, marked as answer]
Yes, you can do that by passing data to the node and implementing a custom node as a react component. You might want to check out [this guide on building custom nodes](https://reactflow.dev/docs/guides/custom-nodes/).
